### PR TITLE
Refactor to support instance configurations

### PR DESCRIPTION
**Pull request by alexs-mparticle**
In cases where we are using multiple instances of the "same" kit,
each instance should maintain its own configuration. For example,
Google Tag Manager should support its own dataLayer variable name
and this should be unique to each instance.

The current implementation will overwrite common with the latest
configuration and will not be unique across mutliple instances.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -289,6 +295,8 @@ var UserAttributeHandler = require('../../../src/user-attribute-handler');
         this.onLogoutComplete = onLogoutComplete;
         this.onModifyComplete = onModifyComplete;
         this.setOptOut = setOptOut;
+
+        this.common = new Common();
```
Comment (rmi22186) [inline index.js]: i don't think this is needed

**Review by rmi22186 — CHANGES_REQUESTED** [review]
i'd like to hold off on reviewing this outside of the one edit i have below until we have the GTM kit fully fleshed out. These should be reviewed in parallel

File: index.js (hunk 1 above)
Comment (alexs-mparticle) [reply index.js]: You're right. It was a relic from something I was experimenting with. I'm going to remove it.

**Review by rmi22186 — APPROVED** [review]
LGTM

**Pull request merged**